tc-health-client/config/config.go

```diff
@@ -328,6 +331,22 @@ func LoadConfig(cfg *Cfg) (bool, error) {
 		if cfg.TOCredentialFile != "" {
 			cfg.CredentialFile.Filename = cfg.TOCredentialFile
 		}
+
+		// if tm-proxy-url is set in the config, verify the proxy
+		// url
+		if cfg.TmProxyURL != "" {
+			if ProxyURL, err = url.Parse(cfg.TmProxyURL); err != nil {
+				ProxyURL = nil
+				return false, errors.New("parsing TmProxyUrl: " + err.Error())
+			}
+			if ProxyURL.Port() == "" {
+				ProxyURL = nil
+				return false, errors.New("TmProxyUrl invalid port specified")
+			}
+			log.Infof("TM queries will use the proxy: %s", cfg.TmProxyURL)
+		} else {
```

Comment:
Nitpick: the `ProxyURL` is already nil, and nothing outside this `if` changes it, so the `else { ProxyURL = nil` isn't necessary

Reply:
@rob05c When the config is reloaded and the tm-proxy-url parameter has been removed, the ProxyURL is set to nil and no proxy will be used the next time a TM is queried.

Reply:
Ah, I see, because `ProxyURL` is a global variable, if `LoadConfig` is called twice it'll be different.
This is why I don't like Global Variables. But yes, you're right, disregard my comment.